checker/src/org/checkerframework/checker/index/upperbound/UBQualifier.java

```diff
@@ -652,16 +668,62 @@ public UBQualifier minusOffset(int value) {
             return addOffset(newOffset);
         }
 
-        private LessThanLengthOf addOffset(OffsetEquation newOffset) {
-            Map<String, Set<OffsetEquation>> plusMap = new HashMap<>(map.size());
-            for (Entry<String, Set<OffsetEquation>> entry : map.entrySet()) {
-                Set<OffsetEquation> plus = new HashSet<>(entry.getValue().size());
-                for (OffsetEquation eq : entry.getValue()) {
-                    plus.add(eq.copyAdd('+', newOffset));
-                }
-                plusMap.put(entry.getKey(), plus);
+        /**
+         * Returns a copy of this qualifier with array-offset pairs where in the original the offset
+         * contains an access of an array length in arrays. The array length access has been removed
+         * from the offset.
+         *
+         * @param arrays access of the length of these arrays are removed
+         * @return Returns a copy of this qualifier with some offsets removed
+         */
```

Comment:
1. an array-offset pair is not required.  If one isn't found than UNKNOWN is returned.  (I'll add this to the Javadoc.)

2.  This method does not return null. OffsetEquation#removeArrayLengths returns null, but computeNewOffsets never returns null.  Would the following be clearer:

````java
     public UBQualifier removeArrayLengthAccess(final List<String> arrays) {
            if (arrays.isEmpty()) {
                return UpperBoundUnknownQualifier.UNKNOWN;
            }
            OffsetEquationFunction removeArrayLen =  new OffsetEquationFunction() {
                        @Override
                        public OffsetEquation compute(OffsetEquation eq) {
                            return eq.removeArrayLengths(arrays);
                        }
                    };
            return computeNewOffsets(removeArrayLen);
        }
````